modules/apigee_edge_actions/modules/apigee_edge_actions_debug/src/EventSubscriber/ApigeeEdgeActionsDebugEventSubscriber.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Copyright 2020 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_actions_debug\EventSubscriber;
+
+use Drupal\apigee_edge_actions\Event\ApigeeEdgeActionsEventInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Implements event subscriber for all apigee_edge_actions events.
+ */
+class ApigeeEdgeActionsDebugEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * ApigeeEdgeActionsDebugEventSubscriber constructor.
+   *
+   * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
+   *   The logger channel.
+   */
+  public function __construct(LoggerChannelInterface $logger) {
+    $this->logger = $logger;
+  }
+
+  /**
+   * Responds to rules events.
+   *
+   * @param \Symfony\Component\EventDispatcher\Event $event
+   *   The event object.
+   * @param string $event_name
+   *   The event name.
+   */
+  public function onRulesEvent(Event $event, $event_name) {
+    // Log the dispatched event.
+    if ($event instanceof ApigeeEdgeActionsEventInterface) {
```

Comment:
I see that rules comes w/it's own logging configuration at `/admin/config/workflow/rules/settings`, but did not see it work when enabled. Is that why we are adding our own logging?

Reply:
Yes this and also Rules does not ship with any logging action. This provides one. Say you want to log a message when a credential is approved..etc.